server/odc-core/src/main/java/com/oceanbase/odc/core/sql/split/SqlSplitter.java

```diff
@@ -833,4 +847,114 @@ private boolean isIdent(int tokenType) {
+        private final StringBuilder buffer = new StringBuilder();
+        private final LinkedList<OffsetString> holder = new LinkedList<>();
+
+        private OffsetString current;
+        private String delimiter;
+        private boolean firstLine = true;
+        private List<String> sqls = new ArrayList<>();
+        private long iteratedBytes = 0;
+
+        private static final int SQL_STATEMENT_BUFFER_SIZE = 5;
+        private static final Character SQL_SEPARATOR_CHAR = '/';
+        private static final Character LINE_SEPARATOR_CHAR = '\n';
+        private static final String SQL_MULTI_LINE_COMMENT_PREFIX = "/*";
+        private static final Set<Character> DELIMITER_CHARACTERS = new HashSet<>(Arrays.asList(';', '/', '$'));
+
+        public SqlStatementIterator(InputStream input, Charset charset, Class<? extends Lexer> lexerType,
+                String delimiter) {
+            this.reader = new BufferedReader(new InputStreamReader(input, charset));
+            this.lexerType = lexerType;
+            this.delimiter = delimiter;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (current == null) {
+                current = parseNext();
+            }
+            return current != null;
+        }
+
+        @Override
+        public OffsetString next() {
+            OffsetString next = current;
+            current = null;
+            if (next == null) {
+                next = parseNext();
+                if (next == null) {
+                    throw new NoSuchElementException("No more available sql.");
+                }
+            }
+            return next;
+        }
+
+        @Override
+        public long iteratedBytes() {
+            return iteratedBytes;
+        }
+
+        private OffsetString parseNext() {
+            try {
+                if (!holder.isEmpty()) {
+                    return holder.poll();
+                }
+                String line;
+                while (holder.isEmpty() && (line = reader.readLine()) != null) {
+                    if (firstLine) {
+                        buffer.append(line);
+                        firstLine = false;
+                    } else {
+                        buffer.append(LINE_SEPARATOR_CHAR).append(line);
+                    }
+                    // SqlSplitter is non-reentrant, so we need to create a new one for each loop
+                    SqlSplitter splitter = createSplitter();
+                    sqls = splitter.split(buffer.toString()).stream().map(OffsetString::getStr)
+                            .collect(Collectors.toList());
+                    while (sqls.size() > SQL_STATEMENT_BUFFER_SIZE) {
+                        String sql = sqls.remove(0);
+                        holder.addLast(new OffsetString(0, sql));
+                        int index = buffer.indexOf(sql.substring(0, sql.length() - 1));
+                        buffer.delete(0, index + sql.length());
```

Comment:
we can not consider that the sql-spliter will not add any str to the sql

Reply:
Actually, SqlSplitter will not add or remove inner a complete SQL statement, It only remove some blank characters.